code/modules/power/supermatter/supermatter.dm

```diff
@@ -841,6 +857,30 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 					to_chat(user, "<span class='notice'>A tiny piece of \the [W] falls off, rendering it useless!</span>")
 			else
 				to_chat(user, "<span class='warning'>You fail to extract a sliver from \The [src]! \the [W] isn't sharp enough anymore.</span>")
+	if(istype(W, /obj/item/crystal_destabilizer))
+		var/obj/item/crystal_destabilizer/injector = W
+		if(!injector.filled)
+			to_chat(user, "<span class='notice'>You already used \the [W]...</span>")
+			return
+		to_chat(user, "<span class='notice'>You carefully begin inject \the [src] with \the [W]... please don't move untill all the steps are finished</span>")
+		if(W.use_tool(src, user, 4 SECONDS, volume=100))
+			to_chat(user, "<span class='notice'>Seems that \the [src] is starting to resonate with the fluid!</span>")
+			playsound(get_turf(src), 'sound/effects/supermatter.ogg', 50, TRUE)
+			if(W.use_tool(src, user, 4.5 SECONDS, volume=100))
+				to_chat(user, "<span class='notice'>The [src] is reacting violently with the fluid!</span>")
+				for(var/i, i<5, i++)
+					src.fire_nuclear_particle()
+					radiation_pulse(src, 550, 4)
+					sleep(5)
+				priority_announce("We detected an unusual spike in the Supermatter Crystal containment protocols, please standby")
+				if(W.use_tool(src, user, 7.5 SECONDS, volume=100))
```

Comment:
ah yes, 7.5 seconds is definitely enough time to react to that priority announcement and stop the guy who's about to cheese an SM delam

Reply:
is not meant to stop the guy from using the object, but to stop the guy from delaminating the crystal